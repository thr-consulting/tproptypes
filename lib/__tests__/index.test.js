import React from 'react';
import {shallow} from 'enzyme';
import {LocalDate} from 'js-joda';
import Money from 'js-money';
import TPropTypes from '..';

describe('Router Location', () => {
	const Comp = props => (
		<div>
			{props.location.hash}
			{props.location.key}
			{props.location.pathname}
			{props.location.search}
		</div>
	);
	Comp.propTypes = {
		location: TPropTypes.location.isRequired,
	};

	it('renders without error', () => {
		expect(shallow(
			<Comp
				location={{
					hash: '#myhash',
					pathname: '/this/is/my/path',
					search: '?var=123',
					state: {},
				}}
			/>,
		)).toMatchSnapshot();
	});
});

describe('RouterHistory', () => {
	const Comp = props => (
		<div>
			{props.history.action}
			{props.history.length}
		</div>
	);
	Comp.propTypes = {
		history: TPropTypes.history.isRequired,
	};

	it('renders without error', () => {
		expect(shallow(
			<Comp
				history={{
					action: 'PUSH',
					length: 15,
				}}
			/>,
		)).toMatchSnapshot();
	});
});

describe('Local Date', () => {
	const Comp = props => (
		<div>
			{props.date.toString()}
		</div>
	);
	Comp.propTypes = {date: TPropTypes.localDate};

	it('renders without error', () => {
		expect(shallow(<Comp date={LocalDate.of(2017, 6, 23)}/>)).toMatchSnapshot();
	});
});

describe('Money', () => {
	const Comp = props => (
		<div>
			{props.money.amount} {props.money.currency}
		</div>
	);
	Comp.propTypes = {money: TPropTypes.money};

	it('renders without error', () => {
		expect(shallow(<Comp money={new Money(500, Money.CAD)}/>)).toMatchSnapshot();
	});
});

describe('String or Number', () => {
	const Comp = props => (
		<div>
			{props.string} {props.num}
		</div>
	);
	Comp.propTypes = {
		string: TPropTypes.stringOrNumber,
		num: TPropTypes.stringOrNumber,
	};

	it('renders without error', () => {
		expect(shallow(<Comp string="blah" num={5}/>)).toMatchSnapshot();
	});
});
