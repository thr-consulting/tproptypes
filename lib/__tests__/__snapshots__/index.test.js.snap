// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Local Date renders without error 1`] = `
<div>
  2017-06-23
</div>
`;

exports[`Money renders without error 1`] = `
<div>
  500
   
  CAD
</div>
`;

exports[`Router Location renders without error 1`] = `
<div>
  #myhash
  /this/is/my/path
  ?var=123
</div>
`;

exports[`RouterHistory renders without error 1`] = `
<div>
  PUSH
  15
</div>
`;
